README.md
# responsive_widget


A new Flutter package for android ,ios which help users toy make the ui more  responsive.


## Screenshot

<div>
<img src="1.png" height="300em" />
<img src="2.png" height="300em" />
<img src="3.png" height="300em" />
</div>
<div>
<img src="4.png" height="300em" />
<img src="5.png" height="300em" />
<img src="6.png" height="300em" />
<img src="7.png" height="300em" />
</div>

### How To Use It


* you can see  the example

["Example"](https://github.com/momagdy194/responsive_widget/blob/main/example/lib/main.dart)


* and add the dependencies to your file.

```yaml
  dependencies:
    flutter:
        sdk:
  responsive_widget:
```


```dart
/// i have make that's class you will wrap your widget with "ResponsiveUi" class which he cant tack parameter 
/// it is type is widget and make instance from Device Size  its name is size///  it will give me type of widget size 
class MyClass extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(body: 
    
    
    ResponsiveUi(builder: (context, size)=> /// that is my class i wrap my widget with it 
    
    ListView(children: [
/// and you can  mange  the size in your widget with replace the 
/// size.screenWidth or ech like that by  size.screenWidth or size.screenHeight
///and you can replace 
///
/// 
      Text('your screen size width',style: TextStyle(fontSize: size.screenWidth)),
      Text('your screen size hight',style: TextStyle(fontSize: size.screenHeight)),
      Text('title',style: TextStyle(fontSize: size.titleText)),///titleText
      Text('subTitle',style: TextStyle(fontSize: size.normalText))///normal

      ,Icon(Icons.error,size: size.normalIconSize)/// normal iconSize;
      ,Icon(Icons.error,size: size.mediumIconSize,)///medium iconSize;
      ,Icon(Icons.error,size: size.bigIconSize,)///big iconSize;
      ,
      Container(height:size.normalButton,child: RaisedButton(onPressed: (){},child: Text("Normal Button",)))///normalButton
      ,SizedBox(height: 15,)
      ,  Container(height:size.largeButton,child: RaisedButton(onPressed: (){},child: Text("Large Button",)))///largeButton


      ,Image(image: AssetImage("assets/images/userImage.png"),height:size.largeImage ,),///large image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.mediumImage ,),///medium image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.smallImage  ,),///small image,


      SizedBox(height: 19,)
    ],) ,));
  }
}
```



This project is a starting point for a Dart
[package](https://pub.dev/documentation/responsive_widget/latest/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our
[online documentation](https://pub.dev/packages/responsive_widget), which offers tutorials,
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
## [2.0.1] - release.

* A new Flutter package for android ,ios which help users to make android and ios responsive ui.



example/example.dart
import 'package:flutter/material.dart';
import 'package:responsive_widget/responsive_ui.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyClass(),
    );
  }
}


class MyClass extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(body: ResponsiveUi(builder: (context, size) =>ListView(children: [

      Text('title',style: TextStyle(fontSize: size.titleText)),///titleText
      Text('subTitle',style: TextStyle(fontSize: size.normalText))///normal

      ,Icon(Icons.error,size: size.normalIconSize)/// normal iconSize;
      ,Icon(Icons.error,size: size.mediumIconSize,)///medium iconSize;
      ,Icon(Icons.error,size: size.bigIconSize,)///big iconSize;
      ,
      Container(height:size.normalButton,child: RaisedButton(onPressed: (){},child: Text("Normal Button",)))///normalButtotn
      ,SizedBox(height: 15,)
      ,  Container(height:size.largeButton,child: RaisedButton(onPressed: (){},child: Text("large Button",)))///largeButton
      ///
      ,Image(image: AssetImage("assets/images/userImage.png"),height:size.largeImage ,),///large image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.mediumImage ,),///medium image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.smallImage  ,),///small image,


      SizedBox(height: 19,)
    ],) ,));
  }
}


lib/responsive_ui.dart
library responsive_ui;

import 'package:flutter/material.dart';
import './src/finctions/get_device_type.dart';
import './src/models/device_info_model.dart';

class ResponsiveUi extends StatelessWidget {
  final Widget Function(BuildContext context, DeviceSize size) builder;

  const ResponsiveUi({Key key, this.builder}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
      builder: (context, constrains) {
        var mediaQueryData = MediaQuery.of(context);
        var size = DeviceSize(
          orientation: mediaQueryData.orientation,
          deviceType: getDeviceType(mediaQueryData),
          screenWidth: mediaQueryData.size.width,
          screenHeight: mediaQueryData.size.height,
          normalText: mediaQueryData.size.width * .04,
          titleText: mediaQueryData.size.width * .067,
          normalIconSize: mediaQueryData.size.width * .08,
          mediumIconSize: mediaQueryData.size.width * .12,
          bigIconSize: mediaQueryData.size.width * .15,
          backGroundIconSize: mediaQueryData.size.width * .22,
          largeImage: mediaQueryData.size.width / 1.5,
          mediumImage: mediaQueryData.size.width / 2,
          smallImage: mediaQueryData.size.width / 4,
          normalButton: mediaQueryData.size.width * .066,
          largeButton: mediaQueryData.size.width * .09,
        );
        return builder(context, size);
      },
    );
  }
}


lib/responsive_widget.dart
library responsive_widget;

import 'package:flutter/material.dart';
import 'package:responsive_widget/src/controller/get_device_type.dart';

import 'src/model/device_info_model.dart';

class ResponsiveUi extends StatelessWidget {
  final Widget Function(BuildContext context, DeviceSize size) builder;

  const ResponsiveUi({Key key, this.builder}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
      builder: (context, constrains) {
        var mediaQueryData = MediaQuery.of(context);
        var size = DeviceSize(
          orientation: mediaQueryData.orientation,
          deviceType: getDeviceType(mediaQueryData),
          screenWidth: mediaQueryData.size.width,
          screenHeight: mediaQueryData.size.height,
          normalText: mediaQueryData.size.width * .04,
          titleText: mediaQueryData.size.width * .067,
          normalIconSize: mediaQueryData.size.width * .08,
          mediumIconSize: mediaQueryData.size.width * .12,
          bigIconSize: mediaQueryData.size.width * .15,
          backGroundIconSize: mediaQueryData.size.width * .22,
          largeImage: mediaQueryData.size.width / 1.5,
          mediumImage: mediaQueryData.size.width / 2,
          smallImage: mediaQueryData.size.width / 4,
          normalButton: mediaQueryData.size.width * .066,
          largeButton: mediaQueryData.size.width * .09,
        );
        return builder(context, size);
      },
    );
  }
}


lib/src/model/device_info_model.dart

import 'package:flutter/material.dart';

import 'enums.dart';

class DeviceSize {
  final Orientation orientation;
  final DeviceType deviceType;
  final double screenWidth;
  final double screenHeight;
  final double localWidth;
  final double localHeight;
  final double normalText;
  final double titleText;
  final double normalIconSize;
  final double mediumIconSize;
  final double bigIconSize;
  final double backGroundIconSize;
  final double largeImage;
  final double mediumImage;
  final double smallImage;
  final double normalButton;
  final double largeButton;

  DeviceSize({
    this.orientation,
    this.deviceType,
    this.screenWidth,
    this.screenHeight,
    this.localWidth,
    this.localHeight,
    this.normalText,
    this.titleText,
    this.normalIconSize,
    this.mediumIconSize,
    this.bigIconSize,
    this.largeImage,
    this.mediumImage,
    this.smallImage,
    this.normalButton,
    this.largeButton,
    this.backGroundIconSize,
  });
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:responsive_widget/responsive_widget.dart';


void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyClass(),
    );
  }
}


class MyClass extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(body: ResponsiveUi(builder: (context, size) =>ListView(children: [

      Text('title',style: TextStyle(fontSize: size.titleText)),///titleText
      Text('subTitle',style: TextStyle(fontSize: size.normalText))///normal

      ,Icon(Icons.error,size: size.normalIconSize)/// normal iconSize;
      ,Icon(Icons.error,size: size.mediumIconSize,)///medium iconSize;
      ,Icon(Icons.error,size: size.bigIconSize,)///big iconSize;
      ,
      Container(height:size.normalButton,child: RaisedButton(onPressed: (){},child: Text("Normal Button",)))///normalButtotn
      ,SizedBox(height: 15,)
      ,  Container(height:size.largeButton,child: RaisedButton(onPressed: (){},child: Text("large Button",)))///largeButton
      ///
      ,Image(image: AssetImage("assets/images/userImage.png"),height:size.largeImage ,),///large image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.mediumImage ,),///medium image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.smallImage  ,),///small image,


      SizedBox(height: 19,)
    ],) ,));
  }
}
